src/ncp-spinel/SpinelNCPTaskJoinerCommission.cpp

```diff
@@ -0,0 +1,182 @@
+/*
+ *
+ * Copyright (c) 2018 Nest Labs, Inc.
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#if HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include "assert-macros.h"
+#include <syslog.h>
+#include <errno.h>
+#include "SpinelNCPTaskJoinerCommission.h"
+#include "SpinelNCPInstance.h"
+#include "any-to.h"
+#include "spinel-extra.h"
+
+using namespace nl;
+using namespace nl::wpantund;
+
+nl::wpantund::SpinelNCPTaskJoinerCommission::SpinelNCPTaskJoinerCommission(
+	SpinelNCPInstance* instance,
+	CallbackWithStatusArg1 cb,
+	bool action,
+	const char *psk,
+	const char *provision_url
+	):	SpinelNCPTask(instance, cb), mAction(action)
+{
+	if (psk)
+	{
+		strncpy(mPsk, psk, sizeof(mPsk));
+		mPsk[strlen(mPsk)] = 0;
+	}
+	if (provision_url)
+	{
+		strncpy(mProvisioningUrl, provision_url, sizeof(mProvisioningUrl));
+		mProvisioningUrl[strlen(mProvisioningUrl)] = 0;
+	}
+}
+
+void
+nl::wpantund::SpinelNCPTaskJoinerCommission::finish(int status, const boost::any& value)
+{
+	SpinelNCPTask::finish(status, value);
+}
```

Comment:
Since we are only calling base class `finish()` here how about not overriding `finish()`?  (inherited method should work)